{{cookiecutter.profile_name}}/slurm-submit.py

```diff
@@ -2,11 +2,34 @@
 """
 Snakemake SLURM submit script.
 """
+import json
+import logging
+import os
+
+import requests
 from snakemake.utils import read_job_properties
 
 import slurm_utils
 from CookieCutter import CookieCutter
 
+logger = logging.getLogger(__name__)
+
+SIDECAR_VARS = os.environ.get("SNAKEMAKE_CLUSTER_SIDECAR_VARS", None)
+DEBUG = bool(int(os.environ.get("SNAKEMAKE_SLURM_DEBUG", "0")))
+
+if DEBUG:
+    logging.basicConfig(level=logging.DEBUG)
+    logger.setLevel(logging.DEBUG)
+
+
+def register_with_sidecar(jobid):
+    sidecar_vars = json.loads(SIDECAR_VARS)
```

Comment:
Add 
```
if SIDECAR_VARS is None:
    return
```
to handle cases where we're not using sidecar (or some other way to handle this case; not calling register_with_sidecar at all?)

Reply:
In this case we would also need to adjust the `config.yaml` to not set `cluster-sidecar`, else snakemake will not work at all?

Reply:
Either that, or throw an error notifying the user that snakemake >=7 is needed. It's always possible to modify the `config.yaml` afterwards, commenting out relevant options. Still, it would be more user-friendly to make the sidecar config an option.